docs/source/compiler.rst

```diff
@@ -113,6 +113,27 @@ the previous example snippet is identical to the following:
     ep = qc.compiler.native_quil_to_executable(np)
     print(ep.program) # here ep is of type PyquilExecutableResponse, which is not always inspectable
 
+Timeouts
+--------
+
+If your circuit is sufficiently complex the compiler may require more time than is permitted by
+default (``10`` seconds). To relax this timeout, you may use the `set_timeout` method:
+
+.. code:: python
+    qc = get_qc(...)
+    qc.compiler.set_timeout(100) # 100 seconds
+
+Prior to version 2.25, process was slightly different. For a QVM
+
+.. code:: python
+    qc = get_qc(..., as_qvm=True)
+    qc.compiler.client.rpc_timeout = 100 # 100 seconds
+
+While for the QPU,
+
+.. code:: python
+    qc = get_qc(..., as_qvm=False)
+    qc.compiler.quilc_client.rpc_timeout = 100 # 100 seconds
```

Comment:
Is it? The instructions I've always used / given is:

On QVM, you need "qc.compiler.client.timeout";
On QPU it is just "qc.compiler.timeout".

Reply:
The timeout is most relevant to the `Client` object that is created. In both `QVMCompiler` and `QPUCompiler` the client is instantiated in `__init__` and so immediately uses the `timeout` arg given to `__init__` (this is the same value that is assigned to `self.timeout` in both classes). Therefore if you want to modify the timeout that is used specifically in quilc, then you have to change `qc.compiler.{client,quilc_client}.rpc_timeout`. `QPUCompiler` has a second client to the translation service, which also uses `self.timeout` but is deferred until the client is requested. I think that means that you probably got it to work by using `qc.compiler.timeout` because the client hadn't yet been created, and so used the new timeout value.

All that to say, this PR doesn't modify the translation service timeout. Should it?

Reply:
Yeah, unless there was a separate well-documented path for the translation service, I'd apply this one timeout in both locations (and say that it applies to both the compiler timeout for quil_to_native_quil, and translation timeout for native_quil_to_executable).

Reply:
Note that a commit I just added now _does_ modify the translation service timeout. That one is set on a per-call basis to the current value of `QPUCompiler#timeout`.